docs/implementing-real-time-reporting.md

```diff
@@ -0,0 +1,106 @@
+---
+title: Implementing real-time reporting
+custom_edit_url: https://github.com/stryker-mutator/mutation-testing-elements/edit/master/docs/implementing-real-time-reporting.md
+---
+
+Real-time reporting was recently implemented in Stryker. This page is created to help other mutation testing frameworks to make use the same functionality.
+
+## How does it work?
+
+Currently, Stryker provides the report with updates using [Server Sent Events](https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events). This was chosen over WebSockets because the dataflow should only be from a mutation framework to the report. The report should not have to send anything to the mutation framework.
+
+## Implementation details
+
+### Preparing the report
+
+First things first, the mutation testing framework should be able to generate a report filled with mutants<sup>1</sup>. These mutants must have the state `Pending`.
```

Comment:
```suggestion
First things first, the mutation testing framework should generate a report with an initial state of the mutants. If the mutants don't yet have a state because they have not been tested you can use the state `Pending`.
```